[WC2-478] Using source_created_at and source_updated_at instead of their server…

madewulf opened the pull request:
See ticket here: https://bluesquare.atlassian.net/browse/WC2-478


The problem is the wrong ordering of visits in this screen
![Screenshot_20240824-143247 (1)](https://github.com/user-attachments/assets/07cafc73-2a81-461b-baab-37a9356ea748)


As a solution: sending source_created_at and source_updated at to the mobile phones in the mobile entity API because we were sending the update and creation date from the server, which hides the timing between visits at WFP, which is problematic because this timing is essential to find out if a beneficiary is late. 


## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are there enough tests


## How to test
Upload and download entity on the mobile app 
## Print screen / video

madewulf committed:
Using source_created_at and source_updated_at instead of their server equivalent for mobile entity api

bramj commented on iaso/api/mobile/entity.py:
```diff
@@ -75,8 +75,14 @@ class Meta:
+    updated_at = serializers.SerializerMethodField()
+
+    def get_created_at(self, instance):
+        return instance.source_created_at.timestamp() if instance.source_created_at else None
```
This would do the trick yes, but maybe fallback to created_at just in case would be best

bramj committed:
WC2-478 Add fallback to created_at/updated_at...

to avoid sending None to the mobile app.

bramj committed:
WC2-478 Enketo: update source_updated_at timestamp...

when updating an instance via Enketo.

bramj committed:
WC2-478 Extract timestamp properties for instances

To facilitate the fetching of source_created/update_at with fallback to
DB timestamps.
Should make it easier to avoid mistakes in the future.

bramj commented:
@madewulf I've added a fix for when a form is edited with Enketo (creation with Enketo was already okay) and I went further to ensure we have the correct timestamp, while not sending None to the mobile app.
I also noticed we needed the same fix on reference instances.
